src/main/java/me/xemor/skillslibrary2/effects/ArrowEffect.java

```diff
@@ -11,15 +11,22 @@ public class ArrowEffect extends Effect implements TargetEffect {
 
     private final double velocity;
     private final int damage;
-    private final EntityType entityType;
+    private final EntityData entityData;
     private final int fireTicks;
     private static final double log099 = Math.log(0.99);
 
     public ArrowEffect(int effect, ConfigurationSection configurationSection) {
         super(effect, configurationSection);
         velocity = configurationSection.getDouble("velocity", 1.0);
         damage = configurationSection.getInt("damage", 4);
-        entityType = EntityType.valueOf(configurationSection.getString("entity", "arrow").toUpperCase());
+
+        if (configurationSection.isString("entity")) {
```

Comment:
This block of code is repeated all over the place, need to work out a way of not pasting this everywhere before we merge. One way would be EntityData.create(parentSection, "theKey", defaultEntityType) that allows passing in only a string.

Reply:
I was under the impression that we weren't going to implement this, but if you want to go for this then I'll implement it. 
I personally think that abetter solution would be to leave as is currently, with the intent to remove in the next breaking change.

Reply:
I think just throw it in ConfigurationData as another option, users of the plugin tend to just LLM hallucinate how they think the config should work, so avoiding punishing them for doing so when their intention is clear is probably better anyway